### PR TITLE
feat/ICL-309 add status to vc

### PR DESCRIPTION
**Pull request by JGiter**
### Contributor checklist

- [x] Breaking changes - check for any existing interfaces changes that are not backward compatible, removed method etc.
- [x] Documentation - document your code, add comments for method, remember to check if auto generated docs were updated.
- [x] Tests - add new or updated existed test for changes you made.
- [x] Migration guide - add migration guide for every breaking change.
- [x] Configuration correctness - check that any configuration changes are correct ex. default URLs, chain ids, smart contract verification on [Volta explorer](https://volta-explorer.energyweb.org/) or [EWC explorer](https://explorer.energyweb.org/).

### CODE REVIEW

File: src/modules/claims/claims.types.ts [hunk 1]
```diff
@@ -176,6 +179,9 @@ export interface IssueClaimRequestOptions {
 
   /** Indicates whether to publish role on-chain or not (default: false) */
   publishOnChain?: boolean;
+
+  /** Allows to track accuracy or relevance of credential */
```
Comment (Harasz) [inline src/modules/claims/claims.types.ts:183]:
```suggestion
  /** Indicates if credential is actual of the time of verification */
```

File: src/modules/claims/claims.service.ts [hunk 2]
```diff
@@ -1312,4 +1325,9 @@ export class ClaimsService {
       await this._signerService.signMessage(arrayify(proofHash))
     );
   }
+
+  /** Sets location of the credential status */
```
Comment (Harasz) [inline src/modules/claims/claims.service.ts:1329]: Complete the documentation

File: src/modules/claims/claims.service.ts [hunk 3]
```diff
@@ -1312,4 +1325,9 @@ export class ClaimsService {
...
+  private _addStatusToCredential(credential: Credential<RoleCredentialSubject>) {
```
Comment (Harasz) [inline src/modules/claims/claims.service.ts:1330]:
```suggestion
  private addStatusToCredential(credential: Credential<RoleCredentialSubject>) {
```

File: src/modules/claims/claims.service.ts [hunk 4]
```diff
@@ -423,6 +425,15 @@ export class ClaimsService {
...
         },
       };
+      if (!credentialStatus) {
+        const credential = this._verifiableCredentialService.createCredential({
+          id: sub,
+          namespace: role,
+          version: version.toString(),
+          issuerFields,
+        });
+        credentialStatus = (await this._addStatusToCredential(credential)).credentialStatus;
```
Comment (Harasz) [inline src/modules/claims/claims.service.ts:435]:
```suggestion
        credentialStatus = (await this.addStatusToCredential(credential)).credentialStatus;
```

Comment (Harasz) [inline src/modules/claims/claims.service.ts:435]:
```suggestion
        const credentialStatus = (await this._cacheClient.addStatusToCredential(credential)).credentialStatus;
```
Do not override incoming parameters especially when it's not a primitive type

Comment (Harasz) [reply src/modules/claims/claims.service.ts:435]: Do we need `addStatusToCredential` method in claim service, we can just use that one from cache client

File: e2e/claim-revocation.e2e.ts [hunk 5]
```diff
@@ -10,12 +15,15 @@ import {
...
   IIssuerDefinition,
   IRevokerDefinition,
+  RoleCredentialSubject,
 } from '../src';
 import { replenish, root, rpcUrl, setupENS } from './utils/setup-contracts';
 import { setLogger } from '../src/config/logger.config';
 import { ConsoleLogger } from '../src/utils/logger';
 import { addressOf } from '@ew-did-registry/did-ethr-resolver';
 
+const {id} = utils;
```
Comment (Harasz) [inline e2e/claim-revocation.e2e.ts:25]:
```suggestion
const { id } = utils;
```

File: src/modules/claims/claims.service.ts [hunk 6]
```diff
@@ -1312,4 +1324,23 @@ export class ClaimsService {
...
     );
   }
+
+  /**
+   * Creates credential status entry in ssi-hub
+   * 
+   * @param param0 role claim data
+   * @returns credential status entry
+   */
+  private async addSsiCredentialStatus({sub, role, version, issuerFields}: AddSsiCredentialStatusOptions  ) {
```
Comment (Harasz) [inline src/modules/claims/claims.service.ts:1334]:
```suggestion
  private async addSsiCredentialStatus({sub, role, version, issuerFields}: AddSsiCredentialStatusOptions): Promise<!type!> {
```

**Review by Harasz — CHANGES_REQUESTED** [review]
It's not gonna work. During credential creation we are generating different UUID and based on UUID SSI-Hub is generating credential status. Fetching credential status should be done in verifiable credential  service during VC creation.

**Comment by JGiter** [comment]
> It's not gonna work. During credential creation we are generating different UUID and based on UUID SSI-Hub is generating credential status. Fetching credential status should be done in verifiable credential service during VC creation.

@Harasz I fixed that. Please look again

File: src/modules/claims/claims.types.ts [hunk 7]
```diff
@@ -374,3 +384,10 @@ export interface ClaimRevocationDetailsResult {
   revoker: string;
   timestamp: number;
 }
+
+export interface AddSsiCredentialStatusOptions {
+  sub: string;
+  role: string;
+  version: number;
+  issuerFields?: IssuerFields[];
+}
```
Comment (Harasz) [inline src/modules/claims/claims.types.ts:393]:
```suggestion
```

File: src/modules/claims/claims.types.ts [hunk 8]
```diff
@@ -1,7 +1,11 @@
 import { utils } from 'ethers';
-import { VerifiablePresentation } from '@ew-did-registry/credentials-interface';
+import {
+  VerifiablePresentation,
+  StatusList2021Entry,
+} from '@ew-did-registry/credentials-interface';
 import { ClaimData } from '../did-registry';
 import { IMessage } from '../messaging/messaging.types';
+import { IssuerFields } from '../verifiable-credentials';
```
Comment (Harasz) [inline src/modules/claims/claims.types.ts:8]:
```suggestion
```

**Review by Harasz — CHANGES_REQUESTED** [review]
Looks good now, just remove unused code

**Review by Harasz — APPROVED** [review]

**Pull request merged**